src/openrct2/entity/Peep.cpp

```diff
@@ -2236,6 +2236,10 @@ static void peep_interact_with_path(Peep* peep, const CoordsXYE& coords)
                         }
                     }
 
+                    auto queueTileCentre = CoordsXY{ CoordsXY{ guest->NextLoc } + CoordsDirectionDelta[guest->PeepDirection] }
+                                               .ToTileCentre();
+                    guest->SetDestination(queueTileCentre);
+
```

Comment:
I have compared RCT2 and OpenRCT2 and I have see that in RCT2 guests are aligned to the center of the tile after joining to the queue, but in OpenRCT2 guests are aligned to the center of the tile at the second tile and not starting from the first tile.

Also without this, guests can still ignore the next guest in the queue if you configure a queue in this way (tested with maximum game speed):
![image](https://user-images.githubusercontent.com/7508197/211124441-2c00598e-7137-4f83-a61f-7c77f1ed7d9e.png)

------

After:
![image](https://user-images.githubusercontent.com/7508197/211124331-f783cef2-3602-4078-b0d2-57f0179cb3a5.png)

Before:
![image](https://user-images.githubusercontent.com/7508197/211124244-37e74389-1415-4817-90cf-f06e1da6de5f.png)

Reply:
This _does_ direct the guest the middle of the tile, in both x and y direction. But if the first queue tile has peeps standing on it, that location will be occupied by other. I don’t know if this is a problem?

Reply:
I do not think that it should be a problem, the logic to update the queue position of a guest takes care about the distance of the next guest in the queue: guest will wait if the distance to the next guest in both axis (X,Y) is less than 7 units no matters if both guests does not have the same direction, I think that this should not interfere with the guest AI (please see my explanation below).

I have done more research and I think that now I have the answer of why guests should be aligned to the middle of the first queue tile.

Sorry for my long text, but I prefer to explain all of my research in order to get more feedback if this issue can be solved in another way or if I have forgot something.

----
When a guest is walking, every time that guest reaches his destination, pathfinding is used to calculate the next tile that guest should walk (it is done by the `CalculateNextDestination()` method at `GuestPathfinding.cpp`) and the destination and direction of the guest is changed to the next tile.

`CalculateNextDestination()` always ends calling `peep_move_one_tile()` and `peep_move_one_tile()` will set the destination of the guest to the middle of the next tile, also `peep_move_one_tile()` will add an offset to the destination to make a spread effect along walking guests that are not queuing.

When a guest joins a queue, his destination will remain having the spread offset because at this point the guest has not yet reached his destination (remember that `CalculateNextDestination()` is only called when a guest reaches his destination), when guest finally reaches his destination pathfinding will change the destination of the guest to the middle of the second queue tile (because guest have the queuing state). [also this is the reason of why guests are aligned to the middle of the second queue tile instead of doing it at the first queue tile]

![image](https://user-images.githubusercontent.com/7508197/211182876-a66c07af-ca91-4443-bd2a-7ef1933b716c.gif)

The spread offset added by `peep_move_one_tile()` is done within a square from the middle of the tile
![image](https://user-images.githubusercontent.com/7508197/211185893-605db3fb-85ff-405c-8d77-39c251063ed5.png)

----
#### Reason of why guests should always be aligned to the middle of the first queue tile:
If you configures a queue in this way:

![image](https://user-images.githubusercontent.com/7508197/211183640-2a4b1266-24f8-4658-9f6f-f83f7843cf40.png)

There is a low chance that a guest can ignore the next guest in queue if certain conditions are meet:
![image](https://user-images.githubusercontent.com/7508197/211185786-28573330-816f-4a8b-8e1a-326510c816c4.gif)

Demonstration in-game, the next guest of brown guest is the purple guest:
![image](https://user-images.githubusercontent.com/7508197/211186870-f0a2d95a-12a2-454c-b612-0b255aa7df04.gif)

These conditions cannot occurrs if guests are aligned to the middle of the first queue tile because both guests will have the same destination, and due to that, guest will not be able to reach the destination before the next guest in the queue because as I mentioned before, guest will wait if is too nearby to the next guest in the queue.

Thank you for reading this! I hope that this explanation can provide more context of the issue.

Reply:
This is some excellent work. Could you provide a super brief summary in a comment in the code at this point. Something like:

```
// Force set centre of tile to prevent issues with guests accidentally skipping the queue
```